<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html lang="en">
<head>
<link rel="stylesheet" href="stylesheets/styles.css">
<link rel="stylesheet" href="stylesheets/pygment_trac.css">
<meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=no">

<meta http-equiv="Content-type" content="text/html; charset=UTF-8">
<meta http-equiv="Content-Style-Type" content="text/css">
<title>LIFE 891-002</title>
</head>
<body>
<p class="menublock"><a href="http://morotalab.org/teaching/teaching.html">Back</a></p>

<h1>LIFE 891-002 Integrating Quantitative and Computational Biology into Life Sciences Research</h1>
<h2>Spring 2018</h2>


<br>
<br>
<h3>Time and Location</h3>
<ul>
  <li>MWF 9:30-10:20am</li>
  <li>Beadle E106</li>
</ul>



<br>
<br>
<h3>Prerequisites</h3>
<ul>
  <li>There are no formal prerequisites for this course.</li>
</ul>





<br>
<br>
<h3>Course Description</h3>
This course explores several key techniques in quantitative and computational biology, including historical perspective and context as well as current applications in life sciences research. Lectures and class discussions are designed to complement both readings from the primary and secondary literature and practical experience involving available software. Assignments will focus on developing an understanding of the quantitative aspects of each technique, with appropriate mathematical, computational, and/or statistical foundations, and oral presentation of data and analytical results. This is an intensive reading and computational course intended to develop your critical quantitative thinking and data skills. Five key research areas, each taught by a faculty expert in the field, will be examined during the course of the semester with examples of applications in the life sciences; these may include but are not limited to the following.

<p>
<ul>
<li> Quantitative Trait Loci mapping </li>
<li> Sequence alignment </li>			
<li> Genome assembly and annotation </li>
<li> Hidden Markov Models and Pathway Analysis </li>
<li> Phylogeny and Taxonomy </li>
<li> Characterizing complex ecological communities </li>
</ul>
</p>
We will use the R statistical language and, on occasion, other software such as Perl/Python, C++, or Matlab to conduct our analyses. These are all available on the clusters at the UNL Holland Computing Center


<br>
<br>
<h3>Texts and Reading Materials</h3>
There is no required text for this course.



<br>
<br>
<h3>Schedule</h3>
<ul>
  <li>2/19 (M): Linear mixed model for GWAS [<a href="http://morotalab.org/life891-2018/day1/day1.html">HTML</a>][<a href="http://morotalab.org/life891-2018/day1/lab1.html">R</a>]</li>
  <li>2/21 (W): Linear mixed model for GWAS </li>
  <li>2/23 (F): Genomic prediction (Guest lecture - Dr. Jinliang Yang) [<a href="http://morotalab.org/life891-2018/day2/day2.html">HTML</a>][<a href="http://morotalab.org/life891-2018/day2/lab2.html">R</a>]</li>
  <li>2/26 (M): Interactive deterministic formulas for genomic prediction [<a href="http://morotalab.org/life891-2018/day3/day3.html">HTML</a>]</li>
  <li>2/28 (W): MeSH enrichment analysis in Bioconductor [<a href="http://morotalab.org/life891-2018/day4/day4.html">HTML</a>][<a href="http://morotalab.org/ESALQ2016/day5/lab5.html">R</a>]</li>
</ul>



</body>
</html>
